### ISSUE TITLE
Error: 'b' format requires -128 <= number <= 127

### ISSUE THREAD

**Issue by pbock** [KaSroka/Toshiba-AC-control#61]
Several people, including myself, are seeing [erratic behaviour with the Home Assistant integration](https://github.com/h4de5/home-assistant-toshiba_ac/issues/208):

```
2024-11-22 20:57:35.328 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [1295685784] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 245, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2802, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2845, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1007, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1079, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 758, in async_handle_set_fan_mode_service
    await self.async_set_fan_mode(fan_mode)
  File "/config/custom_components/toshiba_ac/climate.py", line 211, in async_set_fan_mode
    await self._device.set_ac_fan_mode(feature_list_id)
  File "/usr/local/lib/python3.12/site-packages/toshiba_ac/device/__init__.py", line 285, in set_ac_fan_mode
    await self.send_state_to_ac(state)
  File "/usr/local/lib/python3.12/site-packages/toshiba_ac/device/__init__.py", line 147, in send_state_to_ac
    future_state = ToshibaAcFcuState.from_hex_state(self.fcu_state.encode())
                                                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/toshiba_ac/device/fcu_state.py", line 281, in encode
    encoded = self.ENCODING_STRUCT.pack(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
struct.error: 'b' format requires -128 <= number <= 127
```

I have a suspicion that this is caused by sub-zero temperatures. I’m struggling to debug this because temperatures are currently above zero here, but I see two possible bugs:

1. [The `handle_cmd_heartbeat()` coroutine parses base-16 integers.](https://github.com/KaSroka/Toshiba-AC-control/blob/e8d5bb7d0aa324e09a94489f50acdc2b59e90621/toshiba_ac/device/__init__.py#L131-L136) I’m not sure what the heartbeat data looks like, but assuming temperatures are represented as signed binary integers in hexadecimal representation, -2 degrees would be represented as `"fe"`, which `int("fe", base=16)` would parse as `254`. This would then be passed to [`update_from_hbt()`](https://github.com/KaSroka/Toshiba-AC-control/blob/e8d5bb7d0aa324e09a94489f50acdc2b59e90621/toshiba_ac/device/fcu_state.py#L372-L381) which would set `self._ac_outdoor_temperature = 254`. This is very warm indeed and would cause problems when [trying to encode it as a one-byte signed integer in `ToshibaAcFcuState.encode()`](https://github.com/KaSroka/Toshiba-AC-control/blob/e8d5bb7d0aa324e09a94489f50acdc2b59e90621/toshiba_ac/device/fcu_state.py#L280) that are consistent with what’s being logged.
2. Even if `handle_cmd_heartbeat()` correctly parses sub-zero temperatures, the raw values are directly written to `self._ac_outdoor_temperature`. If I understand correctly, [a temperature of -1 degree is internally represented as 0x7e, not 0xff, which instead is `ToshibaAcFcuState.NONE_VAL_SIGNED`](https://github.com/KaSroka/Toshiba-AC-control/blob/e8d5bb7d0aa324e09a94489f50acdc2b59e90621/toshiba_ac/device/fcu_state.py#L39-L50). This should not cause `ToshibaAcFcuState.encode()` to fail, but it might cause further problems down the line.

**Comment by hennikul** [comment]
I did some debugging here, and it looks to me (based on my limited experience and comments on Github) that this happens when the temperature is around 0 (C in my case). 

I added some debugging output and ran the script, and found that self._ac_outdoor_temperature is set to 253 when I get this error (have only managed to trigger it once with debugging so far). When it doesn't fail, it shows a number that seems to be the correct outdoor temperature.
